### PR TITLE
[SYCL] Add one more ESIMD regression test

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

File: SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+static constexpr const int VL = 4;
+
+sycl::event add(size_t n, sycl::buffer<double, 1> &buf_a,
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp]: This function/kernel is defined, but not called. 'int main()' calls only 'iota'. Should 'add' also be called?

File: SYCL/ESIMD/regression/complex-lib-lin.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//
+// And another one is constructed directly from object files
+//
+// RUN: %clangxx -fsycl -shared %t-lib-sycl.o %t-lib-esimd.o \
+// RUN:  -fsycl-device-code-split=per_kernel -Wl,-soname,%S -o %t-lib-o.so
+//
+// RUN: %clangxx -fsycl %t-test.o %t-lib-a.so -o %t-a.run
+// RUN: %clangxx -fsycl %t-test.o %t-lib-o.so -o %t-o.run
+//
+// FIXME: is there better way to handle libraries loading than LD_PRELOAD?
```
Comment (kbobrovs) [inline SYCL/ESIMD/regression/complex-lib-lin.cpp:32]: Please comment why special handling is needed and simple LD_LIBRARY_PATH is not enough?

File: SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp (hunk 1 above)
Comment (kbobrovs) [reply SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp]: I think so, even though it wasn't part of the original scenario. That would strengthen the test.

File: SYCL/ESIMD/regression/complex-lib-lin.cpp (hunk 2 above)
Comment (AlexeySachkov) [reply SYCL/ESIMD/regression/complex-lib-lin.cpp:32]: The thing is that those `.so` libraries are generated using some temporary names/paths provided by LIT infrastructure: I simply wasn't able to find in the documentation how to reliably get path to a _directory_ with those libraries to use `LD_LIBRARY_PATH`

File: SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp (hunk 1 above)
Comment (AlexeySachkov) [reply SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp]: Fixed in dc06d2d38ea1461354015c5f06e1cc9176dfcc20

File: SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  auto E = Q.submit([&](sycl::handler &H) {
+    sycl::accessor acc_a{buf_a, H, sycl::read_only};
+    sycl::accessor acc_b{buf_b, H, sycl::read_only};
+    sycl::accessor acc_c{buf_c, H, sycl::write_only};
+
+    H.parallel_for(n, [=](sycl::id<1> i) SYCL_ESIMD_KERNEL {
+      using namespace sycl::ext::intel::experimental::esimd;
+      unsigned int offset = i * VL * sizeof(int);
+      simd<int, VL> va;
+      va.copy_from(acc_a, offset);
```
Comment (kbobrovs) [inline SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp]: Nit: here and in few other places - there is a convenience constructor:
```suggestion
      simd<int, VL> va(acc_a, offset);
```

File: SYCL/ESIMD/regression/complex-lib-lin.cpp (hunk 2 above)
Comment (kbobrovs) [reply SYCL/ESIMD/regression/complex-lib-lin.cpp:32]: Can the output directory be forced at compile-time and reused at runtime? E.g.
```
%clangxx -fsycl -shared ... -o %T/lib-a.so
...
%GPU_RUN_PLACEHOLDER LD_LIBRARY_PATH=%T:$LD_LIBRARY_PATH %t-o.run
```

Comment (AlexeySachkov) [reply SYCL/ESIMD/regression/complex-lib-lin.cpp:32]: > Can the output directory be forced at compile-time and reused at runtime? 

The main issue is how to obtain a path to a (preferably) unique directory, which will be automatically removed after tests execution?

There is a `%T` built-in [substitution](https://llvm.org/docs/CommandGuide/lit.html#substitutions), but it is marked as "do not use":

> %T | parent directory of %t (not unique, deprecated, do not use)

**Review by kbobrovs — DISMISSED** [review]

File: SYCL/ESIMD/regression/complex-lib-lin.cpp (hunk 2 above)
Comment (AlexeySachkov) [reply SYCL/ESIMD/regression/complex-lib-lin.cpp:32]: Added info about `%T` into the test source in c18962fa70c8900fd6aa9022cd3f75327ebe137d

File: SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp (hunk 3 above)
Comment (AlexeySachkov) [reply SYCL/ESIMD/regression/Inputs/complex-lib-esimd.cpp]: Applied in c18962fa70c8900fd6aa9022cd3f75327ebe137d

**Review by v-klochkov — APPROVED** [review]

**Comment by AlexeySachkov** [comment]
Failing tests don't seem to be related to the PR, it should be ready for merge now

**Pull request merged**